### PR TITLE
Jaytaph bootsector asm

### PR DESCRIPTION
**Pull request by jaytaph**
This is a raffler written in assembler and fitted into a bootsector, so it runs directly on (minimum 16bit x86 CPUs). Please check the readme for further information.

### CODE REVIEW

**Comment by ramondelafuente** [comment]
Damn sir... you must have been around for a long **long** time.

creepy-old-man

**Comment by jaytaph** [comment]
Pleased to meet you. Hope,you guessed my name.. :smiling_imp:

**Comment by lucasvanlierop** [comment]
Supercool, I tried to run it but I'm doing something wrong I guess. Could you add an `install.sh` file to make life easy for us?

**Comment by jaytaph** [comment]
It is unfortunately not as easy as just setting up a install.sh, since it really depends on the OS you are running. Can you point out what you did, and what went wrong? Maybe I can check if the documentation is ok or not, and fix that (and maybe get that into an install.sh).

**Comment by lucasvanlierop** [comment]
I think I'm using `bochs` the wrong way, I tried `bochs floppy.img`, however since I' surrounded by screaming babies and shit (for real) I currently have no time to dive into the bochs docs.

That said @rosstuck Maybe it's becoming time for Dockerize

**Comment by jaytaph** [comment]
ah.. bochs issues.. yes, i can fix that.. it needs a configuration file..  (or you can use virtualbox as well.. I can write an installer for that with vboxmanage)

**Comment by jaytaph** [comment]
@lucasvanlierop added bochsrc and some docs on how to run it..

**Comment by lucasvanlierop** [comment]
Thanks for the update, I tried again, first it complained about a missing io debug plugin:
```sh
lucasvanlierop@lucas-xps13:~/workspace/rafflers/jaytaph-bootsector-asm$ bochs -f bochsrc 
========================================================================
                       Bochs x86 Emulator 2.4.6
             Build from CVS snapshot, on February 22, 2011
                   Compiled at Jun  8 2013, 05:16:04
========================================================================
00000000000i[     ] LTDL_LIBRARY_PATH not set. using compile time default '/usr/lib/bochs/plugins'
00000000000i[     ] BXSHARE not set. using compile time default '/usr/share/bochs'
00000000000i[     ] reading configuration from bochsrc
00000000000p[     ] >>PANIC<< bochsrc:2: unknown optional plugin 'iodebug'
00000000000e[CTRL ] notify called, but no bxevent_callback function is registered
00000000000i[CTRL ] quit_sim called with exit code 1
```

When I removed that from the config it says this:

```sh
lucasvanlierop@lucas-xps13:~/workspace/rafflers/jaytaph-bootsector-asm$ bochs -f bochsrc 
========================================================================
                       Bochs x86 Emulator 2.4.6
             Build from CVS snapshot, on February 22, 2011
                   Compiled at Jun  8 2013, 05:16:04
========================================================================
00000000000i[     ] LTDL_LIBRARY_PATH not set. using compile time default '/usr/lib/bochs/plugins'
00000000000i[     ] BXSHARE not set. using compile time default '/usr/share/bochs'
00000000000i[     ] reading configuration from bochsrc
00000000000e[     ] bochsrc:32: Bochs is not compiled with internal debugger support
00000000000i[     ] lt_dlhandle is (nil)
00000000000p[     ] >>PANIC<< dlopen failed for module 'x': file not found
========================================================================
Event type: PANIC
Device: [     ]
Message: dlopen failed for module 'x': file not found

A PANIC has occurred.  Do you want to:
  cont       - continue execution
  alwayscont - continue execution, and don't ask again.
               This affects only PANIC events from device [     ]
  die        - stop execution now
  abort      - dump core 
  debug      - hand control to gdb
Choose one of the actions above: [die] 
```

**Comment by jaytaph** [comment]
hm.. the debug is probably because of the fact that I precompiled bochs with some debug options (because of some OS development, it's nice to have a decent gdb-alike debugger out there that can help me)..  the 'x' module is 'display-module' that isn't found. Most likely its running on another display system, but i don't know which one (you can also start bochs without any options, and configure it from the menu. Maybe you have better luck with that?).

Otherwise, try and see if it works from virtualbox? It might be quicker to check that one out?

**Comment by jaytaph** [comment]
image

**Comment by lucasvanlierop** [comment]
Sorry, haven't found much time to check this. Could you add a Docker of Vagrant setup so it just works out of the box?...please?..... :-D

**Comment by jaytaph** [comment]
Docker will never work (this thing doesnt run on any OS).  Vagrant doesn't support floppy images.

But i've manually created a `install.sh` script that compiles the floppy images and a `vboxrun.sh` that can be used to setup automatically a vbox machine (basically, this is the only thing that vagrant does.. only with a bit more cruft :p )

**Comment by jaytaph** [comment]
@lucasvanlierop try again please..

File: jaytaph-bootsector-asm/install.sh [hunk 1]
```diff
@@ -0,0 +1,23 @@
+#!/bin/sh
+
+#
+# This stuff uses linux. Deal with it!
+#
+
+sudo apt-get -yy install dosfstools mtools nasm
+
+# Creates an empty 1.44MB floppy image
+dd if=/dev/zero of=mrfloppy.img bs=512 count=2880
```
Comment (lucasvanlierop) [inline jaytaph-bootsector-asm/install.sh]: `mrfloppy.img` should be `mr_floppy.img`

**Comment by lucasvanlierop** [comment]
Yes sir that helps a lot!, I've discovered one minor typo other that it's cool ;-)

**Comment by lucasvanlierop** [comment]
Well done :+1:  

Sticker van de juf

**Pull request merged**